### PR TITLE
(feature): add user roles

### PR DESCRIPTION
**Pull request by kowal**
Groundwork before adding authorization rules:

* add `cancancan` gem
* add `role` to `AdminUser` model
* updates admin pages for users
* get rid of `update_attributes is deprecated` rails warning
* add missing specs

Pivotal: https://www.pivotaltracker.com/n/projects/2362592/stories/169121017

### CODE REVIEW

File: app/admin/admin_users.rb [hunk 1]
```diff
@@ -72,7 +81,8 @@
   controller do
     # update without password if not provided
     def update_resource(object, attributes)
-      update_method = attributes.first[:password].present? ? :update_attributes : :update_without_password
+      update_method = attributes.first[:password].present? ? :update : :update_without_password
```
Comment (kowal) [inline app/admin/admin_users.rb]: FYI: 'update_attributes' are deprecated in Rails 6 (will be removed in Rails 6.1) - https://github.com/rails/rails/pull/31998.

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    @user = user
+
+    if @user
+      send(@user.role.to_sym)
```
Comment (kowal) [inline app/models/ability.rb]: this is not used yet, will be implemented in https://www.pivotaltracker.com/n/projects/2362592/stories/169121082

File: app/admin/admin_users.rb (hunk 1 above)
Comment (tsubik) [reply app/admin/admin_users.rb]: :+1:

**Comment by simaob** [comment]
rebased and will merge now.

**Pull request merged**